primer/src/Primer/Typecheck.hs

```diff
@@ -744,7 +744,7 @@ check t = \case
             scrutWrap <- Hole <$> meta' (TCSynthed (TEmptyHole ())) <*> pure (addChkMetaT (TEmptyHole ()) e')
             pure $ Case caseMeta scrutWrap [] CaseExhaustive
       Left (TDIPrim tc) -> do
-        unless (tc == tInt || tc == tChar) $ throwError' $ InternalError $ "Unknown primitive type: " <> show tc
+        unless (tc == tInt || tc == tChar || tc == tAnimation) $ throwError' $ InternalError $ "Unknown primitive type: " <> show tc
         let f b = case caseBranchName b of
               PatCon _ -> Nothing
               PatPrim pc -> case pc of
```

Comment:
It might be nice here to link to a note giving an overview of the `Animation` type. In particular to explain why we do not allow pattern matching on it

Reply:
I added this line yesterday to fix a test failure\*, without initially looking at the context, which I was going to ask you about. But anyway, I've worked it out now, and realised the logic was actually wrong, and have now fixed this in 76d51cc.

I've also rewritten the whole passage in a form which I consider much easier to follow, and which would extend more easily to adding further primitives.

\* This can be hit when `cabal run primer-test -- -p "tcWholeProg idempotent" --hedgehog-replay "375/638:oAbA2d Seed 3320906864796586236 8917501983448077275"`

Reply:
Thanks for refactoring this, the new version is indeed much clearer